staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -337,6 +352,22 @@ func (e *Store) ListPredicate(ctx context.Context, p storage.SelectionPredicate,
 
 // Create inserts a new item according to the unique key from the object.
 func (e *Store) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	var cleanup func(context.Context, bool)
+
+	if e.BeginCreate != nil {
+		fn, err := e.BeginCreate(ctx, obj, options)
+		if err != nil {
+			return nil, err
+		}
+		cleanup = fn
+		defer func() {
+			// If we succeed, we will have already set cleanup to nil.
+			if cleanup != nil {
+				cleanup(ctx, false)
```

Comment:
What do we suggest cleanup functions do if they fail to cleanup?

Reply:
Log/Event.  Have an async GC system (like service repair loops).  Unless we want to *really* plumb REAL transactions in here, there's got to be a turtle at the bottom that doesn't return an error.